atlp-rwanda/hackers-ec-be
# #187419058 Reset password via email

#### What does this PR do?
          Enables users who forgot his/her password to reset it and get a new one without creating a new account
#### Description of Task to be completed?

1. A user who forgot their password can request resetting it
2. Sending a reset password email containing a link along with a token to reset the password
3. Reset the password using the provided token
4. Token is used only once even if it is not expired

#### How should this be manually tested?

1. Chrone Repository into your local machine
2. Run `npm install`
3. Fill all environment variables in `.env` file as they appear in .env example file
4. Run `npm run dev`
5. Use Postman to create a user on the endpoint: `api/v1/users/register`
6. Create another request in Postman to request a password reset on the endpoint: `/api/v1/users/forgot-password`
7. Check the message containing the reset link on the email you have provided or the one of the account you are resetting the password for
8.  Add another request for resetting the password by setting the token you have got from the above step in the params on the endpoint: `/api/v1/users/reset-password/:token` and provide a new password in the body
9. Try login with both old and new passwords to see if the reset happened successfully on the endpoint: `/api/v1/users/login`

#### What are the relevant pivotal tracker/Trello stories?

#187419058
